### ISSUE TITLE
wrap_comments can break up type links

### ISSUE THREAD

**Issue by PSeitz** [rust-lang/rustfmt#5260]
Similar to #5250, I have a problem when addressing types. They get split over multiple lines and are not valid links anymore in rust doc.

```
/// [IntermediateBucketResult](crate::aggregation::intermediate_agg_result::IntermediateBucketResult) 
```

This is converted to:
```
/// [IntermediateBucketResult](crate::aggregation::intermediate_agg_result::
/// IntermediateBucketResult) 
```

**Comment by ytmimi** [comment]
Thanks for the report!

We fixed wrapping reference style links back in #5096. As a workaround for now you might consider rewriting the link as:
```rust
/// [IntermediateBucketResult]
/// [IntermediateBucketResult]: crate::aggregation::intermediate_agg_result::IntermediateBucketResult
```

I Imagine we could take a similar approach to the one introduced in that PR and add another regular expression check for the `[bar][Bar]` and `[bar](Bar)` link syntax.

**Comment by PSeitz** [comment]
Thanks, but I'd like to hide the path to the type.
I created a PR to fix this issue
